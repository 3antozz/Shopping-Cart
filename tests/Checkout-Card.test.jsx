import { describe, it, expect, beforeEach, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import CheckoutCard from "../src/Components/Checkout-Card/Checkout-Card.jsx";
import { MemoryRouter } from "react-router-dom";

describe("Checkout Card Component", () => {
    let total = 200;
    let onClick = vi.fn();
        beforeEach(() => {
            render(
            <MemoryRouter>
                <CheckoutCard total={total} onClick={onClick} />
            </MemoryRouter>
            );
        })
    it("renders summary container", () => {
        expect(screen.getByTestId(/checkout-container/i)).toBeInTheDocument();
    })
    it("render total value", () => {
        expect(screen.getAllByText("200 $").length).toBe(2);
    })
    it("calls function when button clicked", async() => {
        const user = userEvent.setup();
        const button = screen.getByRole("button", {name: /checkout/i});
        await user.click(button);
        expect(onClick).toBeCalled();
    })
        it("matches snapshot", () => {
            let total = 200;
            let onClick = vi.fn();
            const {container} = render(
                <MemoryRouter>
                    <CheckoutCard total={total} onClick={onClick} />
                </MemoryRouter>
            );
            expect(container).toMatchSnapshot();
        });
})