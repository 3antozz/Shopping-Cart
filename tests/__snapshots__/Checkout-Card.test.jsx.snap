// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkout Card Component > matches snapshot 1`] = `
<div>
  <div
    class="_card_65c2c8"
    data-testid="checkout-container"
  >
    <div
      class="_info_65c2c8"
    >
      <h3>
        Subtotal:
      </h3>
      <h3>
        200
         $
      </h3>
    </div>
    <div
      class="_info_65c2c8"
    >
      <h3>
        Shipping:
      </h3>
      <h3>
        Free
      </h3>
    </div>
    <div
      class="_info_65c2c8"
    >
      <h2>
        Total:
      </h2>
      <h2>
        200
         $
      </h2>
    </div>
    <button>
      Checkout
    </button>
  </div>
</div>
`;
